crates/dmm-tools/src/dmi.rs

```diff
@@ -60,6 +60,7 @@ pub mod render {
         /// This is here so that duplicate icon states can be handled by not relying on the btreemap 
         /// of state names in [`Metadata`]. 
         pub fn render_state<W: std::io::Write>(&self, icon_state: &State, target: W) -> io::Result<RenderType> {
+            // Note: we don't use [`State::is_animated()`] here because we actually need to destructure the tuples.
```

Comment:
I'm not in love with the fact that these two functions can experience divergence, since that'll break one of IDB's assumptions, but I have no idea how else to do this given the requirements to destructure here.

Reply:
As a countermeasure against broken images in my own code resulting from this possibility, I made it so that IDB errors when this happens. (Since `render_state` returns what it actually used, can compare "what actually happened" to "predicted file extension".)

Reply:
RIP "stable API", the correct solution to this, which I implemented in 3b610a8 and 23dbc72 is a guard pattern to allow the end user to do some things between prepping to render and rendering.